src/Agent.Plugins/Artifact/FileContainerProvider.cs

```diff
@@ -11,6 +11,7 @@
 using Microsoft.VisualStudio.Services.Agent.Util;
 using Microsoft.VisualStudio.Services.BlobStore.Common;
 using Microsoft.VisualStudio.Services.BlobStore.WebApi;
+using Microsoft.VisualStudio.Services.BlobStore.WebApi.Contracts;
```

Comment:
[not related to line] Are the changes compatible without deploying server updates?  The agent needs to be forward and backward compatible with server to support deployment rollout and downlevel server releases

Reply:
Old agent + old server = no issues
Old Agent + new server = Download Build Artifacts v1 task fails to parse hash pointing at content in new domains (content in existing domains unaffected)
New Agent + Old Server + New Domains turned on (controlled via knob or blobstore) = publish failures
New Agent + New Server = no issues